terraform/full_environment/events_ingestion.tf

```diff
@@ -0,0 +1,188 @@
+resource "google_bigquery_table" "search-event" {
+  dataset_id          = google_bigquery_dataset.dataset.dataset_id
+  table_id            = "search-event"
+  project             = var.gcp_project_id
+  schema              = file("./files/search-event-schema.txt")
+  deletion_protection = true
+  time_partitioning {
+    field = "eventTime"
+    type  = "DAY"
+  }
+
+}
+
+# ga4 'select_item' events get transformed and inserted into this time-partitioned search-event table defined with a vertex schema 
+resource "google_bigquery_table" "view-item-event" {
+  dataset_id          = google_bigquery_dataset.dataset.dataset_id
+  table_id            = "view-item-event"
+  project             = var.gcp_project_id
+  schema              = file("./files/view-item-event-schema.txt")
+  deletion_protection = false
+  time_partitioning {
+    field = "eventTime"
+    type  = "DAY"
+  }
+}
+
+# bucket for function .zip
+resource "google_storage_bucket" "storage_analytics_transfer_function" {
+  name     = "storage_analytics_transfer_function"
+  location = var.gcp_region
+}
+
+# zipped function into bucket
+resource "google_storage_bucket_object" "function_zipped" {
+  name   = "func.zip"
+  bucket = google_storage_bucket.storage_analytics_transfer_function.name
+  source = data.archive_file.init.output_path
+}
+
+# archive .py and requirements.txt to zip
+data "archive_file" "init" {
+  type        = "zip"
+  source_dir  = "${path.module}/files/function/"
+  output_path = "${path.module}/files/init.zip"
+}
+
+# gen 2 function for transferring from bq - ga4 to bq - vertex events schema
+### TO DO - starting with just one schema
+resource "google_cloudfunctions2_function" "function_analytics_events_transfer" {
+  name        = "function_analytics_events_transfer"
+  description = "function that will trigger daily transfer of GA4 data within BQ to BQ instance used for search"
+  location    = ""
+  build_config {
+    runtime = "python311"
+    source {
+      storage_source {
+        bucket = google_storage_bucket.storage_analytics_transfer_function.name
+        object = google_storage_bucket_object.function_zipped.name
+      }
+    }
+    environment_variables = {
+      project_name = var.gcp_project_id,
+      dataset_name = google_bigquery_dataset.dataset.dataset_id
+    }
+  }
+  service_config {
+    max_instance_count            = 5
+    vpc_connector_egress_settings = "ALL_TRAFFIC"
+    ingress_settings              = "ALLOW_INTERNAL_ONLY"
+  }
+}
+
+# 
+resource "google_service_account" "trigger_function" {
+  account_id   = "ga4-to-vertex-transfer"
+  display_name = "ga4_to_bq_vertex_transfer"
+  project      = var.gcp_project_id
+}
+
+# custom role for triggering transfer function
+resource "google_project_iam_custom_role" "trigger_function_role" {
+  role_id     = "trigger_function_role"
+  title       = "scheduler_ga4_to_bq_vertex_transfer-permissions"
+  description = "Enables read-only access to the search-api-v2 Rails app"
```

Comment:
This description is copypasta'd I think

Reply:
Another thought just now: doesn't the SA that invokes this function need the BQ permissions? Because right now the `analytics_read` SA doesn't seem to get used for anything.